src/org/omegat/core/data/ExternalTMXEntry.java

```diff
@@ -0,0 +1,125 @@
+               2013 Aaron Madlon-Kay
+               2014 Alex Buloichik, Aaron Madlon-Kay
+               Home page: http://www.omegat.org/
+               Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.data;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.omegat.util.TMXProp;
+
+/**
+ * Storage for TMX entry.
+ *
+ * Variables in this class can be changed only before store to ProjectTMX. After that, all values must be
+ * unchangeable.
+ *
+ * Only RealProject can create and change TMXEntry objects.
+ *
+ * @author Alex Buloichik (alex73mail@gmail.com)
+ * @author Guido Leenders
+ * @author Aaron Madlon-Kay
+ */
+public class ExternalTMXEntry extends TMXEntry {
+    public List<TMXProp> otherProperties;
+
+    ExternalTMXEntry(ITMXEntry from) {
+        super(from);
+
+        this.otherProperties = from.getProperties();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (! super.equals(obj)) {
+            return false;
+        }
+        // Now compare properties
+        try {
+            ITMXEntry other = (ITMXEntry) obj;
+            if (other.getProperties() == null) {
+                return (this.otherProperties == null) || (this.otherProperties.size() == 0);
+            } else if (this.otherProperties == null) {
+                return (other.getProperties() == null) || (other.getProperties().size() == 0);
+            } else if (other.getProperties().size() != this.otherProperties.size()) {
+                return false;
+            } else {
+                // same properties in same order
+                for (int i = 0; i < otherProperties.size(); i++) {
+                    if (! otherProperties.get(i).equals(other.getProperties().get(i))) {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        } catch (ClassCastException cc) {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(changeDate / 1000, creationDate / 1000, translation, note, changer, creator,
+                source);
+    }
+    
+    public List<TMXProp> getProperties() {
+        return otherProperties;
```

Comment:
This should return a copy of properties. For example, 
```java
return Collections.unmodifiableList(otherProperties)
```

Reply:
Agree, change pushed today